ext/crypto/00_crypto.js

```diff
@@ -491,62 +511,185 @@
+
+              // 8.
+              // Section 4.3 of RFC7517
+              if (jwk.key_ops) {
                 if (
-                  normalizedAlgorithm.length > length ||
-                  normalizedAlgorithm.length <= (length - 8)
+                  ArrayPrototypeFind(
+                    jwk.key_ops,
+                    (u) => !ArrayPrototypeIncludes(recognisedUsages, u),
+                  ) !== undefined
                 ) {
                   throw new DOMException(
-                    "Key length is invalid",
+                    "`key_ops` member of JsonWebKey is invalid",
                     "DataError",
                   );
                 }
-                length = normalizeAlgorithm.length;
-              }
 
-              if (keyUsages.length == 0) {
-                throw new DOMException("Key usage is empty", "SyntaxError");
+                if (
+                  !jwk.key_ops.every((u) =>
+                    ArrayPrototypeIncludes(keyUsages, u)
+                  )
+                ) {
+                  throw new DOMException(
+                    "`key_ops` member of JsonWebKey is invalid",
+                    "DataError",
+                  );
+                }
               }
 
-              const handle = {};
-              WeakMapPrototypeSet(KEY_STORE, handle, {
-                type: "raw",
-                data: keyData,
-              });
-
-              const algorithm = {
-                name: "HMAC",
-                length,
-                hash,
-              };
-
-              const key = constructKey(
-                "secret",
-                extractable,
-                usageIntersection(keyUsages, recognisedUsages),
-                algorithm,
-                handle,
-              );
+              // 9.
+              if (jwk.ext == false && extractable == true) {
+                throw new DOMException(
+                  "`ext` member of JsonWebKey is invalid",
+                  "DataError",
+                );
+              }
 
-              return key;
+              break;
             }
-            // TODO(@littledivy): jwk
             default:
               throw new DOMException("Not implemented", "NotSupportedError");
           }
+
+          // 5.
+          let length = data.byteLength * 8;
+          // 6.
+          if (length === 0) {
+            throw new DOMException("Key length is zero", "DataError");
+          }
+          if (normalizeAlgorithm.length) {
+            // 7.
+            if (
+              normalizedAlgorithm.length > length ||
+              normalizedAlgorithm.length <= (length - 8)
+            ) {
+              throw new DOMException(
+                "Key length is invalid",
+                "DataError",
+              );
+            }
+            length = normalizeAlgorithm.length;
+          }
+
+          if (keyUsages.length == 0) {
+            throw new DOMException("Key usage is empty", "SyntaxError");
+          }
```

Comment:
Where does the spec say to do this?

Reply:
https://w3c.github.io/webcrypto/#SubtleCrypto-method-importKey (importKey Step 9)